token_client.go

```diff
@@ -24,57 +24,47 @@ type clientCredentialsManagerConfig struct {
 
 var ErrInvalidCredentials = errors.New("invalid credentials")
 
-func newClientCredentialsTokenManager(config *clientCredentialsManagerConfig) *clientCredentialsTokenManager {
-	return &clientCredentialsTokenManager{
+func newTokenClient(config *TokenClientConfig) *TokenClient {
```

Comment:
This has to become public so that the library's client can create a token provider.